### PR TITLE
Feature email templates payments subscriptions

### PR DESCRIPTION
**Pull request by rwhite27**

### CODE REVIEW

File: api/controllers/PaymentsController.php [hunk 1]
```diff
@@ -176,4 +181,45 @@ protected function sendWebhookResponseEmail(Users $user, array $payload, string
...
+     * @param $user
+     * @param $payload
+     * @return void
+     */
+    public function updateSubscriptionPaymentStatus(Users $user, array $payload): void
+    {
+        $chargeDate = date('Y-m-d H:i:s', $payload['data']['object']['created']);
+
+        //Fetch current user subscription
+        $subscription = Subscription::findFirst([
```
Comment (kaioken) [inline api/controllers/PaymentsController.php]: @rwhite27 make this into a function no, or I believe I already have something to get the current user subscription

File: library/Models/Subscription.php [hunk 2]
```diff
@@ -160,4 +172,23 @@ public static function getActiveForThisApp() : Subscription
...
+    /**
+     * Get subscription by user's default company;
+     * @param Users $user
+     * @return Subscription
+     */
+    public static function getByDefaultCompany(Users $user): Subscription
+    {
+        $subscription = self::findFirst([
+            'conditions' => 'user_id = ?0 and companies_id = ?1 and apps_id = ?2 and is_deleted  = 0',
+            'bind' => [$user->id, $user->default_company, Di::getDefault()->getApp()->getId()]
```
Comment (kaioken) [inline library/Models/Subscription.php]: @rwhite27 $user->getid() and use the metho to get the default company

File: library/Middleware/AclMiddleware.php [hunk 3]
```diff
@@ -41,6 +43,16 @@ public function call(Micro $api)
             // DELETE -> delete
             // POST -> create
 
+            //Search current company's app setting with key paid to verify payment status for current company
+            $subscriptionPaid = CompaniesSettings::findFirst([
+                'conditions' => "companies_id = ?0 and name = 'paid' and is_deleted = 0",
```
Comment (kaioken) [inline library/Middleware/AclMiddleware.php]: @rwhite27 emcapsulate this to a method

File: api/routes/api.php [hunk 4]
```diff
@@ -47,21 +47,16 @@
...
 $router->post('/users/{id}/devices', [
-    'Gewaer\Api\Controllers\UsersController',
+    'Gewaer\Api\Controllers\UserLinkedSourcesController',
     'devices',
 ]);
 
 //detach mobile devices
 $router->post('/users/{id}/devices/{deviceId}/detach', [
-    'Gewaer\Api\Controllers\UsersController',
+    'Gewaer\Api\Controllers\UserLinkedSourcesController',
```
Comment (kaioken) [inline api/routes/api.php]: @rwhite27 just use delete, with the device id no need for the call dettached

Comment (rwhite27) [reply api/routes/api.php]: We can't do that as the detachment process requires an app parameter to find out which source the app belongs to. @kaioken

**Pull request merged**